astropy/io/fits/fitsrec.py

```diff
@@ -71,36 +71,36 @@ def __init__(
 
     def __getitem__(self, key):
         if isinstance(key, str):
-            indx = _get_index(self.array.names, key)
+            index = _get_index(self.array.names, key)
 
-            if indx < self.start or indx > self.end - 1:
+            if index < self.start or index > self.end - 1:
                 raise KeyError(f"Key '{key}' does not exist.")
         elif isinstance(key, slice):
             return type(self)(self.array, self.row, key.start, key.stop, key.step, self)
         else:
-            indx = self._get_index(key)
+            index = self._get_index(key)
 
-            if indx > self.array._nfields - 1:
+            if index > self.array._nfields - 1:
                 raise IndexError("Index out of bounds")
 
-        return self.array.field(indx)[self.row]
+        return self.array.field(index)[self.row]
 
     def __setitem__(self, key, value):
         if isinstance(key, str):
-            indx = _get_index(self.array.names, key)
+            index = _get_index(self.array.names, key)
 
-            if indx < self.start or indx > self.end - 1:
+            if index < self.start or index > self.end - 1:
                 raise KeyError(f"Key '{key}' does not exist.")
         elif isinstance(key, slice):
-            for indx in range(slice.start, slice.stop, slice.step):
-                indx = self._get_indx(indx)
```

Comment:
Here is the suspicious code I mentioned in the opening post. The `_get_indx()` method doesn't seem to exist (at least `git grep` can't find its definition). That being said, I am not familiar enough with this code to know how to trigger this bug, so I don't know how to write the regression test. It is also possible that this is dead code not a bug. It's best if the `io.fits` maintainers (@saimn, @astrofrog) have a look at this.

Reply:
It seems fairly obvious this piece of code is bad -- at the very least it should use `range(*slice.indices(<some length>))` but agreed that it is best to just go bug-for-bug here.

Reply:
This was a real bug and the fix has already been merged (which also created a merge conflict for this pull request).